src/routes/settings.go

```diff
@@ -421,26 +433,39 @@ func SettingsRoutes(router *gin.Engine, title string, database *db.Database, _bl
 		c.SecureJSON(http.StatusOK, gin.H{"status": "success"})
 	})
 	router.POST("/settings/database/exportSnapshot", func(c *gin.Context) {
-		importDB := host.GetDataDir() + "yourplace.db"
 		exportDB := host.GetDataDir() + "yourplace.db.snapshot"
-		host.DeleteIfExists(exportDB)
-		host.CopyFile(importDB, exportDB)
-		err := db.SanitizeDatabase(exportDB)
-		if err != nil {
-			core.LogDebug("Error sanitizing database: " + err.Error())
-			c.SecureJSON(http.StatusInternalServerError, gin.H{"status": "failure"})
-			return
-		}
+		database.MetaUpdateValue("exportSnapshotStatus", "running")
+		go func() {
+			host.DeleteIfExists(exportDB)
+			err := database.ExportSnapshot(exportDB)
+			if err != nil {
+				core.LogDebug("Error sanitizing database: " + err.Error())
+				database.MetaUpdateValue("exportSnapshotStatus", "failed")
+				return
+			}
+			database.MetaUpdateValue("exportSnapshotStatus", "complete")
+		}()
 		c.SecureJSON(http.StatusOK, gin.H{"status": "success", "exportPath": exportDB})
 	})
 	router.POST("/settings/database/importSnapshot", func(c *gin.Context) {
 		importPath := host.GetDataDir() + "yourplace.db.snapshot"
-		// todo halt indexer prior to import
-		err := database.ImportSnapshot(importPath)
-		if err != nil {
-			c.SecureJSON(http.StatusBadRequest, gin.H{"status": "failed", "error": err.Error()})
-			return
-		}
+		database.MetaUpdateValue("importSnapshotStatus", "running")
+		blockchain.IndexerStop()
+		go func() {
+			for i := 0; i < 100; i++ {
+				uuids := database.IndexerGetRunningJobsUUIDs()
+				if len(uuids) == 0 {
+					break
```

Comment:
can we double check this loop logic? are there other conditions where it needs to exit before the 100 iterations?